bundle/QueryType/SearchQueryType.php

```diff
@@ -27,38 +31,77 @@ public static function getName(): string
 
     protected function configureOptions(OptionsResolver $optionsResolver): void
     {
-        $optionsResolver->setRequired(['search_text', 'content_types', 'subtree']);
-
+        $optionsResolver->setRequired(['search_text', 'content_types', 'subtree', 'sort', 'order']);
         $optionsResolver->setAllowedTypes('search_text', 'string');
         $optionsResolver->setAllowedTypes('content_types', 'string[]');
+        $optionsResolver->setAllowedTypes('sort', 'string[]');
         $optionsResolver->setAllowedTypes('subtree', ['int', 'string']);
-
         $optionsResolver->setAllowedValues(
             'search_text',
             static fn (string $searchText): bool => trim($searchText) !== '',
         );
-
+        $optionsResolver->setAllowedValues(
+            'sort',
+            /** @var string[] $keys */
+            fn (array $keys): bool => $this->sortKeysAllowed($keys),
+        );
+        $optionsResolver->setAllowedValues(
+            'order',
+            static fn (string $searchText): bool => trim($searchText) === 'asc' || trim($searchText) === 'desc',
+        );
         $optionsResolver->setDefault('content_types', $this->configResolver->getParameter('search.content_types', 'ngsite'));
         $optionsResolver->setDefault('subtree', $this->site->getSettings()->rootLocationId);
+        $optionsResolver->setDefault('sort', [Query\SortClause\DatePublished::class]);
+        $optionsResolver->setDefault('order', 'desc');
     }
 
     protected function doGetQuery(array $parameters): Query
     {
         $subtreeLocation = $this->site->getLoadService()->loadLocation($parameters['subtree']);
-
+        $sortingKeys = $parameters['sort'];
+        $order = $parameters['order'];
+        $descendingOrder = $order === 'desc';
         $criteria = [
             new Criterion\Subtree($subtreeLocation->pathString),
             new Criterion\Visibility(Criterion\Visibility::VISIBLE),
         ];
-
         if (count($parameters['content_types']) > 0) {
             $criteria[] = new Criterion\ContentTypeIdentifier($parameters['content_types']);
         }
-
         $query = new LocationQuery();
         $query->query = new FullText(trim($parameters['search_text']));
         $query->filter = new Criterion\LogicalAnd($criteria);
+        $sortClauses = [];
+        foreach ($sortingKeys as $sortingKey) {
+            $sortClauses[] = $this->createSortClause($sortingKey, $descendingOrder);
+        }
+        $query->sortClauses = $sortClauses;
 
         return $query;
     }
+
+    /**
+     * @param string[] $keys
+     *
+     * @return bool
+     */
+    private function sortKeysAllowed(array $keys): bool
+    {
+        foreach ($keys as $key) {
+            if (!class_exists($key)) {
```

Comment:
We should also check if the values are instances of `Ibexa\Contracts\Core\Repository\Values\Content\Query\SortClause` with: `is_a($key, SortClause::class, true)`

Reply:
Fixed!